lib/ansible/executor/module_common.py

```diff
@@ -1334,13 +1350,21 @@ def get_action_args_with_defaults(action, args, defaults, templar):
         module_defaults = templar.template(module_defaults)
 
         # deal with configured group defaults first
-        if action in C.config.module_defaults_groups:
-            for group in C.config.module_defaults_groups.get(action, []):
-                tmp_args.update((module_defaults.get('group/{0}'.format(group)) or {}).copy())
+        for group_name in collection_groups:
+            if 'group/%s' % group_name not in module_defaults:
+                continue
+
+            for collection_name in collection_groups[group_name]:
+                collection_routing = _get_collection_metadata(collection_name).get('action_groups', {})
```

Comment:
var-name bikeshed:

```suggestion
                action_group = _get_collection_metadata(collection_name).get('action_groups', {})
```

Reply:
I guess this one's not as obvious because of the "pre-chewed" format, so `action_group` isn't the right name for the current impl

Reply:
Maybe `action_name_group_map`?

Reply:
I just did `action_group` for now.